PolyEngine/Core/Src/Path.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "Defines.hpp"
+#include "Dynarray.hpp"
+#include "String.hpp"
+
+namespace Poly {
+
+	class CORE_DLLEXPORT Path
+	{
+	public:
+		static Path* Build();
+
+		Path& RootAt(const String& root);
+		Path& Dir(const String& dir);
+		Path& File(const String& file);
+
+		bool IsAbsolute() const { return Root.GetLength(); }
+		bool IsDir() const { return !Filename.GetLength(); }
+
+		String Get();
+
```

Comment:
I'd love some additional convenience methods, like e.g. `::New(startingValue)`, `::SetExtension()`, `operator+`. Also getters for root, path without filename, file stem, file extension, etc.